src/pe_reports/metrics.py

```diff
@@ -15,22 +24,552 @@ def __init__(self, start_date, end_date, org_uid):
+                    "name_server",
+                ]
+            ]
+            domain_sum = domain_sum[:25]
+            domain_sum = domain_sum.rename(
+                columns={
+                    "domain_permutation": "Domain",
+                    "ipv4": "IPv4",
+                    "ipv6": "IPv6",
+                    "mail_server": "Mail Server",
+                    "name_server": "Name Server",
+                }
+            )
+        else:
+            domain_sum = pd.DataFrame(
+                columns=[
+                    "Domain",
+                    "IPv4",
+                    "IPv6",
+                    "Mail Server",
+                    "Name Server",
+                ]
+            )
+        return domain_sum
+
+    def utlds(self):
+        """Return count of unique top level domains."""
+        mal_df = self.df_mal
+
+        if mal_df > 0:
+            mal_df["tld"] = (
+                mal_df["domain_permutation"]
+                .str.split(".")
+                .str[-1]
+                .str.split("/")
+                .str[0]
+            )
+            utlds = len(mal_df["tld"].unique())
+        else:
+            utlds = 0
+
+        return utlds
+
+
+class Malware_Vulns:
+    """Malware and Vulnerabilities Class."""
+
+    def __init__(self, start_date, end_date, org_uid):
+        """Initialize Shodan vulns and malware class."""
+        self.start_date = start_date
+        self.end_date = end_date
+        self.org_uid = org_uid
+        insecure_df = query_shodan(
+            org_uid,
+            start_date,
+            end_date,
+            "shodan_insecure_protocols_unverified_vulns",
+        )
+        self.insecure_df = insecure_df
+
+        vulns_df = query_shodan(org_uid, start_date, end_date, "shodan_verified_vulns")
+        vulns_df["port"] = vulns_df["port"].astype(str)
+        self.vulns_df = vulns_df
+
+        assets_df = query_shodan(org_uid, start_date, end_date, "shodan_assets")
+        self.assets_df = assets_df
+
+    @staticmethod
+    def isolate_risky_assets(df):
+        """Return risky assets from the insecure_df dataframe."""
+        insecure = df[df["type"] == "Insecure Protocol"]
+        insecure = insecure[
+            (insecure["protocol"] != "http") & (insecure["protocol"] != "smtp")
+        ]
+        risky_assets = insecure[["ip", "protocol"]].drop_duplicates(keep="first")
+        return risky_assets
+
+    def protocol_count(self):
+        """Return a count for each insecure protocol."""
+        risky_assets = self.isolate_risky_assets(self.insecure_df)
+        # Horizontal bar: insecure protocol count
+        pro_count = risky_assets.groupby(["protocol"], as_index=False)["protocol"].agg(
+            {"id_count": "count"}
+        )
+        return pro_count
+
+    def risky_assets(self):
+        """Return dataframe of ip and risky protocols."""
+        risky_assets = self.isolate_risky_assets(self.insecure_df)
+        return risky_assets
```

Comment:
Is this function really necessary if all it does is call `self.isolate_risky_assets()` and return its output?

Reply:
@dav3r, we can probably remove this and just call the static method (isolate_risky_assets) instead, as long as we are ok to call the static method while using the class

Reply:
I'm fine with that approach.

Reply:
@dav3r addressed suggestion in the following commit: [6858068](https://github.com/cisagov/pe-reports/pull/99/commits/68580680776f164eb2e8b144afc603029093c0a5)

Reply:
Thanks!